src/aeb_fate_asm.erl

```diff
@@ -572,7 +578,16 @@ deserialize_type(<<6, Rest/binary>>) ->
     {V, Rest3} = deserialize_type(Rest2),
     {{map, K, V}, Rest3};
 deserialize_type(<<7, Rest/binary>>) ->
-    {string, Rest}.
+    {string, Rest};
+deserialize_type(<<8, Size, Rest/binary>>) ->
+    {Variants, Rest2} = deserialize_variants(Size, Rest, []),
+    {{variant, Size, Variants}, Rest2}.
```

Comment:
Why do you need the `Size` here? Isn't it always `length(Variants)`?

Reply:
True. Just thought that when checking this type it would be nice to check that Tag < Size first without calculating length. But not strictly needed. (in the decoded format.)